router_netconsole.c

```diff
@@ -29,6 +29,12 @@
 #include "router_images.h"
 #include "router_types.h"
 
+static const unsigned int ap121f_ip = 3232235777UL; /* 192.168.1.1 */
+static const unsigned int my_ip = 3232235778UL;  /* 192.168.1.2 */
+
+static const uint8_t zero_mac[] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+static const uint8_t ap121f_mac[] = { 0x00, 0x03, 0x7f, 0x09, 0x0b, 0xad };
```

Comment:
This is not the ap121f_mac but an atheros fallback mac in their u-boot. It is used whenever the actual mac address is not properly loaded by the HW manufacturer in the board specific code. Maybe you should rename this to reflect that.

Reply:
interesting...does it mean the manufacturer should make sure the MAC address is written at the proper position so that uboot can fetch it?

Renaming might make sense, but this sounds like other random (different) routers may have the same MAC in the future due to the same problem, right? This would fool the router detection logic.

Reply:
This depends on their initialization code. But yes, they should make sure that they fetch the correct mac address from the flash.

And yes, other routers tend to have the same mac address (due to this problem). If you have some GL-iNet devices (AR750 in my example) then you get the same mac address.